pkg/app/carbonapi/http_handlers.go

```diff
@@ -939,39 +940,148 @@ func (app *App) findHandler(w http.ResponseWriter, r *http.Request, lg *zap.Logg
 		logLevel = zapcore.ErrorLevel
 		return
 	}
-
+	// TODO: extend writeResponse w/new formats and migrate to it
 	if contentType == jsonFormat && jsonp != "" {
-		w.Header().Set("Content-Type", contentTypeJavaScript)
-		if _, writeErr := w.Write([]byte(jsonp)); writeErr != nil {
-			toLog.HttpCode = 499
-			logLevel = zapcore.WarnLevel
-			return
-		}
-		if _, writeErr := w.Write([]byte{'('}); writeErr != nil {
+		if writeResponse(ctx, w, blob, jsonFormat, jsonp) != nil {
 			toLog.HttpCode = 499
 			logLevel = zapcore.WarnLevel
 			return
 		}
+	} else {
+		w.Header().Set("Content-Type", contentType)
 		if _, writeErr := w.Write(blob); writeErr != nil {
 			toLog.HttpCode = 499
 			logLevel = zapcore.WarnLevel
 			return
 		}
-		if _, writeErr := w.Write([]byte{')'}); writeErr != nil {
-			toLog.HttpCode = 499
-			logLevel = zapcore.WarnLevel
-			return
+	}
+
+	toLog.HttpCode = http.StatusOK
+}
+
+func expandEncoder(globs dataTypes.Matches, leavesOnly string, groupByExpr string) ([]byte, error) {
+	var b bytes.Buffer
+	groups := make(map[string][]string)
+	seen := make(map[string]struct{})
+	nodeCount := len(strings.Split(globs.Name, "."))
+	names := make([]string, 0, len(globs.Matches))
+	for _, g := range globs.Matches {
+		if leavesOnly == "1" && !g.IsLeaf {
+			continue
+		}
+		name := g.Path
```

Comment:
would it be more intuitive to call it `path := g.Path`?

Reply:
Sorry, but no. Too many changes.

Reply:
I mean then I need to rename names to paths, and then we'll get `groups[globs.Name] = paths`

Reply:
or I can do that in only 2 lines, if it's more pleasant to your eye, @timtofan :)
```
		path := g.Path
		nodes := strings.SplitN(path, ".", nodeCount+1)
```

Reply:
No, it cause more issues y=then it solves, sorry. :)
Let's keep it as is.